engine/swarm/swarm_manager_locking.md

```diff
@@ -0,0 +1,153 @@
+the ability to protect the encryption key at rest by _autolocking_ your swarm.
+
+When Docker restarts or whenever the encryption key needs to be loaded from disk
+to a node's memory, you must
+[unlock the swarm](swarm_manager_locking.md#unlock-a-swarm) first, using a key
+generated by Docker when the swarm was locked. When a new node joins the swarm,
+the is propagated to it over mutual TLS. You can rotate this key at any time.
+
+## Initialize a swarm with autolocking enabled
+
+When you initialize a new swarm, you you can use the `--autolock` flag to
+automatically lock swarm manager nodes.
+
+```bash
+$ docker swarm init --autolock
+
+Swarm initialized: current node (k1q27tfyx9rncpixhk69sa61v) is now a manager.
+
+To add a worker to this swarm, run the following command:
+
+    docker swarm join \
+    --token SWMTKN-1-0j52ln6hxjpxk2wgk917abcnxywj3xed0y8vi1e5m9t3uttrtu-7bnxvvlz2mrcpfonjuztmtts9 \
+    172.31.46.109:2377
+
+To add a manager to this swarm, run 'docker swarm join-token manager' and follow the instructions.
+
+To unlock a swarm manager after it restarts, run the `docker swarm unlock`
+command and provide the following key:
+
+    SWMKEY-1-WuYH/IX284+lRcXuoVf38viIDK3HJEKY13MIHX+tTt8
+```
+
+Record the key in a safe place.
+
+When Docker restarts, a new manager node attempts to join the swarm, or Docker
+needs to load the encryption key from disk to a node's memory, you need to
+[unlock the swarm](swarm_manager_locking.md#unlock-a-swarm). You will see an
+error like the following and services will not start.
+
+```bash
+$ sudo service docker restart
+
+$ docker service ls
+
+Error response from daemon: Swarm is encrypted and needs to be unlocked before it can be used. Please use "docker swarm unlock" to unlock it.
+```
+
+## Enable or disable autolock on an existing swarm
+
+To enable autolock on an existing swarm, set the `autolock` flag to `true`.
+
+```bash
+$ docker swarm update --autolock="true"
+
+Swarm updated.
+To unlock a swarm manager after it restarts, run the `docker swarm unlock`
+command and provide the following key:
+
+    SWMKEY-1-+MrE8NgAyKj5r3NcR4FiQMdgu+7W72urH0EZeSmP/0Y
+
+Please remember to store this key in a password manager, since without it you
+will not be able to restart the manager.
+```
+
+To disable autolock, set `--autolock` to `false`. The mutual TLS key and the
+encryption key used to read and write RAFT logs will be stored unencrypted on
+disk, so this configuration is not recommended.
+
+```bash
+$ docker swarm update --autolock="false"
+```
+
+Keep the unlock key around for a short time after disabling autolocking, in case
+a manager goes down while it is still configured to lock using the old key.
+
+## Unlock a swarm
+
+To unlock a locked swarm, use `docker swarm unlock`.
+
+```bash
+$ docker swarm unlock
+
+Please enter unlock key:
+```
+
+Enter the encryption key that was emitted when you locked the swarm, and the
```

Comment:
Well, it was generated but it was also emitted to standard output... let me see what I can do.